### PR TITLE
Feature/165 reset password api

### PR DESCRIPTION
**Pull request by uiboy**

### CODE REVIEW

File: openbook_auth/views.py [hunk 1]
```diff
@@ -384,3 +383,54 @@ def get(self, request, user_username):
...
+        email = data.get('email')
+        new_password = data.get('new_password')
+
+        User = get_user_model()
+        user = User.get_user_with_email(email)
+
+        with transaction.atomic():
+            user.verify_password_reset_token(token=token, password=new_password)
+
+        return Response('Password set successfully', status=status.HTTP_200_OK)
```
Comment (lifenautjoe) [inline openbook_auth/views.py]: Return `ApiMessageResponse` and translate the responses with `_` so it shows up in the frontend.

File: openbook_common/tests/helpers.py [hunk 2]
```diff
@@ -23,6 +23,11 @@ def make_fake_post_text():
     return fake.text(max_nb_chars=settings.POST_MAX_LENGTH)
 
 
+def make_fake_username():
+    username = fake.text(max_nb_chars=settings.USERNAME_MAX_LENGTH)
```
Comment (lifenautjoe) [inline openbook_common/tests/helpers.py]: there is a `fake.user_name` function. Text collides many times in tests as its an `n` number of words

File: openbook_auth/models.py [hunk 3]
```diff
@@ -1645,16 +1680,56 @@ def _get_default_follow_lists(self):
...
+            )
+        except jwt.ExpiredSignatureError:
+            raise ValidationError(
+                _('Token expired')
+            )
+        except jwt.DecodeError:
+            raise ValidationError(
+                _('Failed to decode token')
+            )
+        except KeyError:
```
[the remark this answers is not in the document]
Comment (uiboy) [reply openbook_auth/models.py]: If we create a valid token but that doesnt have one of the keys we expect... there is a 500 error django throws a KeyError exception .. i tried to use the token from change_email to reset the password to see if it works and got key error.

File: openbook_common/tests/helpers.py (hunk 2 above)
Comment (uiboy) [reply openbook_common/tests/helpers.py]: I changed it -  though how do we ensure it stays within our username constraints.. i see everywhere in tests just a fake.user_name is used..

File: openbook_auth/models.py [hunk 4]
```diff
@@ -62,6 +62,12 @@ class User(AbstractUser):
     )
 
     uuid = models.UUIDField(default=uuid.uuid4, editable=False, unique=True)
+    JWT_TOKEN_TYPE_CHANGE_EMAIL = 'CE'
+    JWT_TOKEN_TYPE_PASSWORD_RESET = 'PR'
+    JWT_TOKEN_TYPES = (
```
Comment (lifenautjoe) [inline openbook_auth/models.py]: What does this do?

Comment (uiboy) [reply openbook_auth/models.py]: Oh no, that was the idea but tuples dont work that way for constants.. so now i just use two constants. Forgot to remove this.

**Pull request merged**